compass/landice/tests/greenland/decomposition_test/__init__.py

```diff
@@ -39,7 +50,43 @@ def validate(self):
         """
         variables = ['thickness', 'normalVelocity']
         steps = self.steps_to_run
-        if '1proc_run' in steps and '8proc_run' in steps:
-            compare_variables(test_case=self, variables=variables,
-                              filename1='1proc_run/output.nc',
-                              filename2='8proc_run/output.nc')
+        name1 = '{}proc_run'.format(self.cores_set[0])
+        name2 = '{}proc_run'.format(self.cores_set[1])
+        if name1 in steps and name2 in steps:
+            # validate thickness
+            variable = ['thickness', ]
+            if self.velo_solver == 'FO':
+                l1_norm = 1.0e-11
+                l2_norm = 1.0e-11
+                linf_norm = 1.0e-11
+                quiet = False
+            else:
+                l1_norm = 0.0
+                l2_norm = 0.0
+                linf_norm = 0.0
+                quiet = True
+            compare_variables(test_case=self, variables=variable,
+                              filename1='{}/output.nc'.format(name1),
+                              filename2='{}/output.nc'.format(name2),
+                              l1_norm=l1_norm, l2_norm=l2_norm,
+                              linf_norm=linf_norm, quiet=quiet)
+
+            # validate normalVelocity
+            variable = ['normalVelocity', ]
+            if self.velo_solver == 'FO':
+                l1_norm = 1.0e-13
+                l2_norm = 1.0e-15
+                linf_norm = 1.0e-16
+                quiet = False
+            else:
+                l1_norm = 0.0
+                l2_norm = 0.0
+                linf_norm = 0.0
+                quiet = True
+            compare_variables(test_case=self, variables=variable,
+                              filename1='{}/output.nc'.format(name1),
+                              filename2='{}/output.nc'.format(name2),
+                              l1_norm=l1_norm, l2_norm=l2_norm,
+                              linf_norm=linf_norm, quiet=quiet)
```

Comment:
For 2 variables, this format is okay.  I think if you were validating more variables, you would want an outer ` if self.velo_solver == 'FO':` and then you would have a single call to `compare_variables()` for SIA cases and multiple calls for FO cases.  You also wouldn't need to define `quiet` because it is always either `True` or `False`.

Reply:
That's a good suggestion.  I'll take that approach if we encounter this situation again.